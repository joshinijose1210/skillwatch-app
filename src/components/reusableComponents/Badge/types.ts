export type Props = {
    children?: React.ReactNode;
    bgColor?: string;
    pd?: string;
    borderRadius?: string;
};
