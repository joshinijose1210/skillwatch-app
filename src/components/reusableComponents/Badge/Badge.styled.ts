import styled from 'styled-components';
import { Props } from './types';

export const BadgeWrapper = styled.div<Props>`
    background-color: ${({ bgColor }) => bgColor ?? '#fff'};
    border-radius: ${({ borderRadius }) => borderRadius ?? '0.2rem'};
    padding: ${({ pd }) => pd ?? '0.5rem 1rem'};
`;
