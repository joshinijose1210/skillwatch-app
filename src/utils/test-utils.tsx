import { ToastContainer } from '@medly-components/core';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import { store as rtkStore, setupStore } from '@slice';
import { RenderOptions, RenderResult, render } from '@testing-library/react';
import { defaultTheme } from '@theme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import reduxMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { userMockState } from '../mocks/userMockState';
import { initialState } from '../slice';

interface CustomOptions extends RenderOptions {
    initialState?: any;
    initialEntries?: any;
}

interface WithThemeProviderProps {
    children?: React.ReactNode;
}

const mockStore = reduxMockStore(),
    store = mockStore({
        ...initialState,
        user: { ...initialState.user, groups: ['admin'] }
    });

const WithThemeProvider: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <ThemeProvider theme={defaultTheme}>
        <>
            <ToastContainer position="top-end" />
            {children}
        </>
    </ThemeProvider>
);

const WithStore: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <WithThemeProvider>{children}</WithThemeProvider>
    </Provider>
);

const WithRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <MemoryRouter>
        <WithThemeProvider>{children}</WithThemeProvider>
    </MemoryRouter>
);

const WithStoreAndRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <MemoryRouter>
            <WithThemeProvider>{children}</WithThemeProvider>
        </MemoryRouter>
    </Provider>
);

export const RTKRender = (ui: React.ReactElement, { initialState = userMockState, ...renderOptions }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);

    const RtkWrapper: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
        <Provider store={store}>
            <MemoryRouter initialEntries={renderOptions.initialEntries}>
                <WithThemeProvider>{children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
    return { rtkStore, ...render(ui, { wrapper: RtkWrapper, ...renderOptions }) };
};
const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithThemeProvider, ...options });

export const renderWithStore = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStore, ...options });

export const renderWithRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithRouter, ...options });

export const renderWithStoreAndRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStoreAndRouter, ...options });

export const wrapper = (props: { children: React.ReactNode }, { initialState = userMockState }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);
    return (
        <Provider store={store}>
            <MemoryRouter>
                <WithThemeProvider>{props.children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render, store };
