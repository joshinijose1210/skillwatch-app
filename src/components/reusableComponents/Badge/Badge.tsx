import { FC, memo } from 'react';
import { BadgeWrapper } from './Badge.styled';
import { Props } from './types';

export const Badge: FC<Props> = memo(({ children, ...props }) => {
    return <BadgeWrapper {...props}>{children}</BadgeWrapper>;
});

Badge.displayName = 'Badge';
