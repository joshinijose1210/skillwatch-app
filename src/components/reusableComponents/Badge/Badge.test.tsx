import { RTKRender } from '@utils/test-utils';
import { Badge } from './Badge';

describe('Badge', () => {
    it('should render properly', () => {
        const { container } = RTKRender(<Badge />);
        expect(container).toMatchSnapshot();
    });
});
