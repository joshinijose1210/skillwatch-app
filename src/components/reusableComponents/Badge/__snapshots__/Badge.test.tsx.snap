// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render properly 1`] = `
<div>
  <div
    class="sc-1exx2co-0 dufstT"
  />
</div>
`;
